src/main/java/com/vmware/vipclient/i18n/messages/service/ProductService.java

```diff
@@ -84,23 +76,18 @@ public ProductService(BaseDTO dto) {
     /**
      * Retrieves the list of locales of a product. It recursively applies data source fallback mechanism in case of failure.
      *
-     * @param msgSourceQueueIter Iterator of DataSourceEnum sources
      * @return list of locales of the product specified in the dto object
      */
-    public List<String> getSupportedLocales(Iterator<DataSourceEnum> msgSourceQueueIter) {
-        if (!msgSourceQueueIter.hasNext()) { return null; }
-
-        DataSourceEnum dataSource = msgSourceQueueIter.next();
-        ProductOpt opt = dataSource.createProductOpt(dto);
-        List<String> locales = opt.getSupportedLocales();
-        // If failed to get components from the data source
-        if (locales.isEmpty()) {
-            // Try the next dataSource in the queue
-            if (msgSourceQueueIter.hasNext()) {
-                locales = getSupportedLocales(msgSourceQueueIter);
-                // If no more data source in queue, log the error. This means that neither online nor offline fetch succeeded.
-            } else {
-                logger.error(FormatUtils.format(ConstantsMsg.GET_COMPONENTS_FAILED, dataSource.toString()));
+    public List<String> getSupportedLocales(){
+        List<String> locales = null;
+        Iterator<DataSourceEnum> msgSourceQueueIter = VIPCfg.getInstance().getMsgOriginsQueue().iterator();
+        while((locales == null || locales.isEmpty()) && msgSourceQueueIter.hasNext()){
+            DataSourceEnum dataSource = msgSourceQueueIter.next();
+            ProductOpt opt = dataSource.createProductOpt(dto);
+            locales = opt.getSupportedLocales();
+            // If failed to get locales from the data source, log the error.
+            if (locales.isEmpty()) {
```

Comment:
```suggestion
            if (locales==null || locales.isEmpty()) {
```

Reply:
Actually ‘locales’ won't be null as opt.getSupportedLocales() method new ArrayList and won't return null.